Hm 8 User 도메인 개발

GlenJeong opened the pull request.

GlenJeong committed:
HM-8 user 도메인 개발

GlenJeong committed:
HM-8 user 도메인 개발 그리고 도메인 TDD

goyounha11 commented:
디벨롭 풀 받고 다시 PR오픈해주세요 다 컨플릭트 나고 있네요

GlenJeong committed:
HM-8 user 도메인 개발 그리고 도메인 TDD(충돌 해결 완료)

GlenJeong commented:
충돌 완료 했습니다.

goyounha11 commented:
@GlenJeong 공백 컨벤션에 대한 부분 다시 검토 후 멘션주세요

GlenJeong committed:
HM-8 User 도메인 불필요한 공백 삭제 및 줄 정렬, 공백 컨벤션

GlenJeong commented:
User 도메인 불필요한 공백 삭제하고 줄 정렬 및 공백 컨벤션 했습니다.

GlenJeong commented:
Develop에서 내려 받고 난 후 일부 코드 오류 난 거 정리 
	공통 소스와 중복 된 부분 중 내 소스에서 삭제 
	core -> preconditions 삭제
	util -> RegexType, RegexUtil 삭제
	config -> SecurityConfiguration 삭제 
	testFixtures -> userFixture -> fixture 패키지로 이동
-----------------------------------------------------------------

User 도메인 불필요한 공백 삭제 및 줄 정렬, 공백 컨벤션
	              날짜 validate 추가 및 사용하지 않는 코드 삭제
UserTest 회원 가입, 회원 수정, 회원 탈퇴 테스트 코드 진행

-----------------------------------------------------------------

utils
RegexType -> birthDay 추가
RegexUtil -> 날짜 정규식 추가

core -> exception -> ExceptionCode -> NOT_FOUND_USER, NOT_DATE_FORM 추가


Develop에서 pull하고 난 뒤 현재까지 작업 내역 간단히 올립니다.

goyounha11 commented on src/main/java/com/hwamok/api/UserController.java:
```diff
@@ -0,0 +1,65 @@
+
+    @PostMapping // 회원 가입
+    public ResponseEntity<ApiResult<?>> signup(@RequestBody UserCreateDto.Request request) {
+        userService.create(
```
해당 메소드가 회원가입이라는 것을 명시하고 싶다면 JAVA doc을 써주세요

GlenJeong replied:
불필요한 주석 삭제 했습니다.

goyounha11 commented on src/main/java/com/hwamok/api/UserController.java:
```diff
@@ -0,0 +1,65 @@
+    public ResponseEntity<ApiResult<User>> withdrawUser(@PathVariable long id) {
+        userService.withdraw(id);
+        User user = userService.getUser(id);
+        return Result.ok(user);
```
탈퇴한 유저를 반환해주는 이유가 있나요?

GlenJeong replied:
아니요  삭제했습니다.

goyounha11 commented on src/main/java/com/hwamok/api/dto/AddressCreateDto.java:
```diff
@@ -0,0 +1,22 @@
+
+    }
+
+}
```
공백 컨벤션 지켜주세요.

GlenJeong replied:
네 컨벤션 적용했습니다.

goyounha11 commented on src/main/java/com/hwamok/api/dto/AddressUpdateDto.java:
```diff
@@ -0,0 +1,22 @@
+
+    }
+
+}
```
여기도요

GlenJeong replied:
적용했습니다.

goyounha11 commented on src/main/java/com/hwamok/api/dto/UploadedFileCreateDto.java:
```diff
@@ -0,0 +1,19 @@
+      private String savedFileName;
+
+    }
+}
```
여기도요

GlenJeong replied:
적용했습니다

goyounha11 commented on src/main/java/com/hwamok/api/dto/UploadedFileUpdateDto.java:
```diff
@@ -0,0 +1,19 @@
+      private String savedFileName;
+
+    }
+}
```
여기도요

GlenJeong replied:
적용했습니다

goyounha11 reviewed (changes requested):
@GlenJeong 공백 컨벤션 확인 다시 해주세요

GlenJeong committed:
HM-8 User 도메인 개발 추가 post에 대한 validate 및 PreConditions
테스트 코드 추가 post가 음수인 경우와 5자리 아래일 경우
Dto 컨벤션 적용
UserController - 탈퇴 유저 정보 반환 삭제
	           - 의미 없는 주석 삭제

goyounha11 commented on src/main/java/com/hwamok/core/exception/ExceptionCode.java:
```diff
@@ -14,7 +14,9 @@ public enum ExceptionCode {
+    NOT_POST_FORM("E009", "우편번호 형식이 다릅니다.")
+
 
     ;
```
공백제거해주세요

GlenJeong replied:
네 알겠습니다.

goyounha11 commented on src/main/java/com/hwamok/user/service/UserServiceImpl.java:
```diff
@@ -12,32 +12,41 @@
     @Override
-    public void withdraw(long id) {
+    public void delete(long id) {
         User user = userRepository.findById(id).orElseThrow(() -> new RuntimeException("NOT_FOUND_USER"));
```
유저가 탈퇴하는 행위는 WIthdraw가 어울리지않나요?
바꾸신 이유가 뭘까영

GlenJeong replied:
withdraw보다 delete 의미가 더 명확하게 전달 되는 것 같아서 변경했습니다. 다시 바꿀까요?

goyounha11 replied:
@GlenJeong 비지니스 로직에 가깝게 서비스 메소드명을 지어야합니다.
디비 입장에서는 삭제하는 행위일지 몰라도 비지니스 로직이 탈퇴라면 탈퇴가 더 적당할듯합니다.

GlenJeong replied:
네 맞네요~ 비즈니스 로직에 맞게 메소드명을 변경했습니다.

GlenJeong committed:
HM-8 User 도메인 개발 ExceptionCode에 의미 없는 공백 삭제 및 주석 제거

changgyu-kim commented on src/main/java/com/hwamok/api/UserController.java:
```diff
@@ -0,0 +1,68 @@
+@RequestMapping("/user")
+@RequiredArgsConstructor
+public class UserController {
+
```
저희 컨벤션은 클래스아래 공백없이 코드를 작성해가기로 했었습니다.  다른곳도 붙여서 작성해주세요

GlenJeong replied:
네 수정했습니다.

changgyu-kim commented on src/main/java/com/hwamok/api/dto/user/AddressCreateDto.java:
```diff
@@ -0,0 +1,17 @@
+import lombok.NoArgsConstructor;
+
+public class AddressCreateDto {
+
```
여기도 붙여야 합니다

GlenJeong replied:
네 수정했습니다.

changgyu-kim commented on src/main/java/com/hwamok/user/domain/Address.java:
```diff
@@ -1,14 +1,36 @@
+  @Column(length = 5, nullable = false)
   private int post;
   private String addr;
   private String detailAddr;
```
주소와 상세주소 length 설정도 필요하지 않을까요?

GlenJeong replied:
네 필요해 보이네요~ 최대 주소 길이를 예측해 보고 설정해 보도록 하겠습니다.

GlenJeong replied:
네 의견 수렴하여 주소와 상세주소 length 설정해서 수정했습니다.

changgyu-kim commented on src/main/java/com/hwamok/user/domain/JoinPlatform.java:
```diff
@@ -2,8 +2,35 @@
+  SELF("S001", "자사 회원"),
+  KAKAO("K001", "카카오 회원"),
+  NAVER("N001", "네이버 회원"),
+  GOOGLE("Goo1", "구글 회원");
```
기존에 작성했던 코드에서 code와 message를 추가한 이유가 있을까요? 다른곳에서 사용되어지지는 않는 것 같습니다

GlenJeong replied:
필요하지 않을 까 싶어서 작성했는 데 현재까지는 쓸 일이 없는 듯 해요 불필요한 코드로 보이니 지웠다가 필요한 시점이 생기면 추가할게요~

GlenJeong replied:
네 의미없는 코드 삭제 후 수정했습니다.

changgyu-kim commented on src/test/java/com/hwamok/user/domain/UserTest.java:
```diff
@@ -0,0 +1,573 @@
+    // - given : 준비
+    // - when : 실행
+    // - then : 결과
+    // 3. 결이 다를 때
```
불필요한 주석입니다

GlenJeong replied:
네~ 지울게요

GlenJeong replied:
의미 없는 주석 삭제했습니다.

changgyu-kim commented on src/test/java/com/hwamok/user/domain/UserTest.java:
```diff
@@ -0,0 +1,573 @@
+        assertThat(user.getAddress().getAddr()).isEqualTo("15, Deoksugung-gil, Jung-gu, Seoul, Republic of Korea");
+        assertThat(user.getAddress().getDetailAddr()).isEqualTo("201");
+    }
+
```
회원탈퇴 메서드는 user의 status만 변경되는데 이것만 테스트 해보면 되지 않을까요?

GlenJeong replied:
네 맞네요 수정했습니다

changgyu-kim commented on src/test/java/com/hwamok/user/domain/UserTest.java:
```diff
@@ -0,0 +1,573 @@
+                        "01012345678", "GOOGLE","originalImage", "savedImage",
+                        12345, "15, Deoksugung-gil, Jung-gu, Seoul, Republic of Korea", "201"));
+    }
+
```
email은 50글자까지 작성가능한걸로 구현하였는데 실패사유를 50글자 초과로 쓰셔야 할 거 같습니다. 다른 코드도 비슷한 점이 있으면 수정해주세요

GlenJeong replied:
네 맞네요 의견 수렴했습니다

changgyu-kim commented on src/main/java/com/hwamok/user/service/UserServiceImpl.java:
```diff
@@ -0,0 +1,52 @@
+@Transactional
+@RequiredArgsConstructor
+public class UserServiceImpl implements UserService {
+
```
여기도 공백 삭제해주세요

GlenJeong replied:
네 삭제 했습니다.

changgyu-kim commented on src/main/java/com/hwamok/user/service/UserServiceImpl.java:
```diff
@@ -0,0 +1,52 @@
+                        originalFileName, savedFileName, post, addr, detailAddr));
+
+        return user;
+    }
```
user객체가 다른로직을 수행 안하는것 같은데 따로 값을 받아서 리턴하지 말고 바로값을 리턴해주면 코드가 더 깔끔할 것 같습니다. 비슷한 다른코드도 해당됩니다.

GlenJeong replied:
네 좋은 의견 수렴했습니다.

changgyu-kim commented on src/main/java/com/hwamok/user/service/UserServiceImpl.java:
```diff
@@ -0,0 +1,52 @@
+
+        user.delete();
+
+        userRepository.save(user);
```
@Transaction 에노테이션이 있으므로 세이브할 필요는 없어보입니다.

GlenJeong replied:
상태 변경이 일어나면 트랜잭션이 커밋될 때 자동으로 DB에 반영되니 save 메소드는 의미가 없어 보이네요 수정했습니다.

changgyu-kim commented on src/main/java/com/hwamok/api/UserController.java:
```diff
@@ -0,0 +1,68 @@
+        return Result.created();
+    }
+
+    @PatchMapping("/update/{id}")
```
@PatchMapping에 update의미가 있는데 api에 update를 사용하는것은 불필요해 보입니다

GlenJeong replied:
네 의견 수렴하여 수정했습니다.

changgyu-kim commented on src/main/java/com/hwamok/api/UserController.java:
```diff
@@ -0,0 +1,68 @@
+        return Result.ok(user);
+    }
+
+    @DeleteMapping ("/delete/{id}")
```
@DeleteMapping에 delete의미가 있는데 api 에 delete를 사용하는것은 불필요해 보입니다

GlenJeong replied:
네 의견 수렴하여 수정했습니다.

GlenJeong committed:
HM-8 User 도메인 개발 추가 핸드폰 정규식 및 테스트

GlenJeong committed:
HM-8 User 도메인 개발 수정 불필요한 주석 삭제, 컨벤션 적용, 의미없는 코드 삭제

changgyu-kim reviewed (approved):
수고하셧습니다~

GlenJeong committed:
HM-8 User 도메인 개발 비지니스 로직에 맞게 서비스 메소드명

goyounha11 reviewed (approved):
고생하셨습니다.

GlenJeong committed:
Merge branch 'develop' into HM-8
